nucliadb_protos/knowledgebox.proto

```diff
@@ -21,6 +21,40 @@ message KnowledgeBoxID {
     string uuid = 2;
 }
 
+// External Index node provider
+enum ExternalIndexProviderType {
```

Comment:
I think we could think about nucliadb's index node as an index provider too, so we don't have to think about an abstraction only for Pinecone.
Therefore, the `UNSET` become `NUCLIA` or `NUCLIADB`.